### ISSUE TITLE
Analyzer with Options

### ISSUE THREAD

**Issue by paul1956** [dotnet/roslyn#5840]
The available VB analyzer templates create 3 projects (a PCL, test and VSIX) but if your analyzer needs to interact with the IDE for options, you get send down a deep rabbit hole. There needs to be a template for VSIX/NuGet projects that want to allow user "options" which only creates 2 projects and possibility a menu item. All the code needs to be in the VSIX to be able to interact with Visual Studio and allow the user to configure options. Plus a BKM for handling options should also be included in template or documentation. If there is no interest in creating the templates, warning about the above should be included somewhere. What is the point of adding buttons when the VXIX code can't interact with the PCL analyzers.

**Comment by kanekotic** [comment]
I completely agree, is a headache how to add options to an analyzer. Is there any current recommended way?. This are the options that are going around my mind.
- I am trying to go trough the root of an additionalFile in the destination project, but this wont work for a VSIX. 
- There should be a way to access the settings maybe as other VSIX configuration  the WritableSettingsStore could be an option,
- a local file to the VSIX export and the nuget package, so it can be access in the same way, this solves the problem of a common solution but makes it more difficult to the user to modify/ .

**Comment by CyrusNajmabadi** [comment]
Not planning on adding templates here.
